src/System.Private.CoreLib/src/System/Threading/Thread.CoreCLR.cs

```diff
@@ -312,23 +299,12 @@ private void SetStartHelper(Delegate start, int maxStackSize)
         private extern void StartupSetApartmentStateInternal();
 #endif // FEATURE_COMINTEROP_APARTMENT_SUPPORT
 
-        /// <summary>Retrieves the name of the thread.</summary>
-        public string Name
+        partial void ThreadNameChanged()
         {
-            get => _name;
-            set
+            string value = Name;
```

Comment:
This does not work same way as before in subtle ways.

Reply:
That would be for the null case which is odd but it's same as before. What else am I missing?

Reply:
The name is cached on the unmanaged side too. When we set the name on the managed side outside the lock, the name on managed and unmanaged sides can get out of sync.